core/src/main/java/org/apache/iceberg/rest/HTTPClient.java

```diff
@@ -186,8 +186,21 @@ private static void throwFailure(
     throw new RESTException("Unhandled error: %s", errorResponse);
   }
 
-  private URI buildUri(String path, Map<String, String> params) {
+  private String buildBaseUri(String path) {
+    // if full path is provided, use the input path as baseUri
+    if (path.startsWith("https://")) {
```

Comment:
what if it's a `http` endpoint?

Reply:
I thought it would be unsafe for an auth server endpoint to be http - but I'm fine to leave that judgment to the users